src/aws-cpp-sdk-core/include/smithy/tracing/TracingUtils.h

```diff
@@ -0,0 +1,97 @@
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+#pragma once
+
+#include <aws/core/monitoring/HttpClientMetrics.h>
+#include <smithy/Smithy_EXPORTS.h>
+#include <smithy/tracing/Meter.h>
+#include <functional>
+#include <chrono>
+#include <utility>
+
+namespace smithy {
+    namespace components {
+        namespace tracing {
+            static const char SMITHY_METRICS_DNS_DURATION[] = "smithy.client.http.dns_duration";
+            static const char SMITHY_METRICS_CONNECT_DURATION[] = "smithy.client.http.connect_duration";
+            static const char SMITHY_METRICS_SSL_DURATION[] = "smithy.client.http.ssl_duration";
+            static const char SMITHY_METRICS_THROUGHPUT[] = "smithy.client.http.throughput";
+            static const char SMITHY_METRICS_UNKNOWN_METRIC[] = "smithy.client.http.unknown_metric";
+
+            class SMITHY_API TracingUtils {
+            public:
+                TracingUtils() = default;
+
+                template<typename T>
+                static T MakeCallWithTiming(std::function<T()> func,
+                    Aws::String metricName,
+                    std::shared_ptr<Meter> meter,
+                    Aws::Map<Aws::String, Aws::String> attributes,
```

Comment:
if you just forward-move attributes, it worth checking if you can pass them by an r-value and std::forward next.